letsencrypt/client/network2.py

```diff
@@ -280,26 +321,19 @@ def answer_challenge(self, challb, response):
         try:
             authzr_uri = response.links['up']['url']
         except KeyError:
-            raise errors.NetworkError('"up" Link header missing')
+            # TODO: Right now Boulder responds with the authorization resource
+            # instead of a challenge resource... this can be uncommented
+            # once the error is fixed.
+            return None
+            # raise errors.NetworkError('"up" Link header missing')
         challr = messages2.ChallengeResource(
             authzr_uri=authzr_uri,
             body=messages2.ChallengeBody.from_json(response.json()))
         # TODO: check that challr.uri == response.headers['Location']?
         if challr.uri != challb.uri:
-            raise errors.UnexpectedUpdate(challr.uri)
+            raise errors.UnexpectedUpdate(challb.uri)
```

Comment:
I think the word "including" is confusing here and indeed might lead to belief that the spec is contradicting itself. I understood this as "The server provides a 200 response containing (only) the updated challenge resource", which would make it consistent with the diagram from the spec. Since you are able to answer challenges separately, I would expect the response to be symmetrically "separate", i.e. response should be Challenge Resource instead of Authorization Resource. Validate status won't be available immediately, but "acknowledgement" status could be available immediately.

letsencrypt/acme-spec#105
https://github.com/letsencrypt/boulder/issues/130